internal/sysfs/osfile.go

```diff
@@ -98,9 +98,32 @@ func (f *osFile) reopen() (errno experimentalsys.Errno) {
 	// Clear any create flag, as we are re-opening, not re-creating.
 	f.flag &= ^experimentalsys.O_CREAT
 
+	var (
+		offset int64
+		err    error
+	)
+
+	if isDir, errno := f.IsDir(); errno == 0 && !isDir {
+		offset, err = f.file.Seek(0, 1)
+		if err != nil {
+			return experimentalsys.UnwrapOSError(err)
+		}
+	}
+
 	_ = f.close()
 	f.file, errno = OpenFile(f.path, f.flag, f.perm)
-	return
+	if errno != 0 {
+		return errno
+	}
+
+	if isDir, errno := f.IsDir(); errno == 0 && !isDir {
+		_, err = f.file.Seek(offset, 0)
```

Comment:
```suggestion
		_, err := f.file.Seek(offset, 0)
```

use the short form here